### ISSUE TITLE
Base URL config Regression

### ISSUE THREAD

**Issue by chrishoage** [dagu-org/dagu#718]
I pulled main and noticed base_url was no longer looking. 

Taking a peek at recent commits  https://github.com/dagu-org/dagu/pull/717 reverted the changes in global.go which allowed this to work. 

Curious why those changes were reverted? Were they causing some other issue? The PR had no body so it's difficult to say why this change was reverted. 

In any case `DAGU_BASE_PATH` no longer functions 

Steps to Reproduce:

1. Build dagu container `sudo docker build -t dagu .` 
2. Run with out `DAGU_BASE_PATH` set `docker run -p "127.0.0.1:8080:8080" --rm -it dagu dagu start-all`
3. observe dagu works 
4. Run with `DAGU_BASE_PATH` set `docker run --env "DAGU_BASE_PATH=/dagu" -p "127.0.0.1:8080:8080" --rm -it dagu dagu start-all`
5. hit `localhost:8080` 
6. observe no redirect to DAGU_BASE_PATH
7. observe when navigating to  `localhost:8080/dagu` app does not load
8. observe request to `/dagu/assets/bundle.js` returns the index 

Revering the commit from  https://github.com/dagu-org/dagu/pull/717 allows all cases in above steps to work properly

**Comment by yohamta** [comment]
My apologies for the regression introduced. I've reverted the PR and am now investigating the issue with the base path and Nginx proxy.

While Dagu works correctly when accessed directly at http://localhost:8080/dagu, I'm still encountering problems when proxying requests from http://localhost/dagu to that location using Nginx. I suspect the issue lies within the Nginx configuration and how it handles the base path. I'll be looking into this further and will provide an update once I've identified a solution. Any suggestions or insights on configuring Nginx for base paths are welcome.

**Comment by chrishoage** [comment]
Could you perhaps explain in more detail about what issues you are seeing with nginx? 

The following works for me (using my original commit) 

```yaml
# docker-compose.yaml
services:
  nginx:
    image: nginx:alpine
    container_name: nginx
    hostname: nginx
    ports:
      - "8081:80"
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
  caddy:
    image: caddy
    container_name: caddy
    hostname: caddy
    ports:
      - "8080:80"
    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile
  dagu:
    image: dagu # build and tag from the local repo using `docker build -t dagu .`
    container_name: dagu
    hostname: dagu
    environment:
      - DAGU_BASE_PATH=/dagu
```


```nginx
# nginx.conf
server {
  listen 80;

  server_name dagu;

  location /dagu {
    proxy_pass http://dagu:8080;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }
}
```

```Caddyfile
# Caddyfile
http://localhost {
  handle /dagu* {
    reverse_proxy http://dagu:8080
  }
}
```

Finally running `docker compse up` 

Hitting  both `http://localhost:8080/dagu` and `http://localhost:8081/dagu` yield Caddy and NGINX servers respectively

**Comment by yohamta** [comment]
Thank you so much for sharing the detailed configurations. I discovered the issue was with my local setup, and everything is working perfectly now. Apologies for the inconvenience!

**Comment by chrishoage** [comment]
Fixed in 678dd21a05aeb968ffbe9aac01a5b6d46026a232
